miden-tx/src/testing.rs

```diff
@@ -0,0 +1,523 @@
+use alloc::vec::Vec;
```

Comment:
The tests were moved inside their own module to fix clippy unused warnings when building the library
Because the tests are now inside a tests module, clippy complained about the nested `tests::tests`, so the top-level was renamed to testing, matching the feature name
The `MockDataStore` was made public, so that it can be used from the integration tests, and the divergence in code from the integration tests and this was resolved